### ISSUE TITLE
Issues with shared formulas in tables during Save

### ISSUE THREAD

**Issue by paintballkev812** [EPPlusSoftware/EPPlus#95]
I have come across an issue with saving an excel sheet with shared formulas. During the save process, EPPlus appears to convert the shared formula to individual cell formulas causing inconsistent values between Excel only and EPPlus excel files.

I've been able to recreate the issue by creating a simple table with 2 data columns and 1 calculation column. The calc column involves using the value in the previous row during the calculation so the first row has its value changed to 0. By simply opening and saving the file, EPPLUS seems to convert the shared formula to individual formulas which causes cell C2 value to change from '0' to the previously shared formula, resulting in a '#VALUE'. 

For testing purposes, I tried the same process in closedXML and lightSpreadsheet and these did not exhibit the issue, so it seems to the way EPPlus is handling things. It appears Excel's priority is Formula>Value>Shared Formula, which is why the value of '0' shows even though the table column has a shared formula. When EPPLUS converts this shared formula to individual formulas during the SaveAs() process, this causes the value to change. 

I think the fix would be to have EPPlus to not convert the shared formulas during the SaveAs process. If that is a limitation, maybe there should be some logic to prevent this from happening (don't assign the shared formula to the individual cell if the cell's value isn't empty?)

[before.xlsx](https://github.com/EPPlusSoftware/EPPlus/files/4529339/before.xlsx)
![Before](https://i.imgur.com/JrBCeHH.png)
[after.xlsx](https://github.com/EPPlusSoftware/EPPlus/files/4529342/after.xlsx)
![After](https://i.imgur.com/2ljjtA3.png)

Here is the code that is being used:
```
xlPackage = New ExcelPackage(New IO.FileInfo(SaveFileDialog1.FileName))
xlPackage.SaveAs(New IO.FileInfo(SaveFileDialog1.FileName))
```

**Comment by JanKallman** [comment]
EPPlus will set the formula for the entire column upon save (from the table settings). In this case the first row is set to 0 (fixed), causing the invalid value in cell C2.
I'll try to add a fix for this shortly.

**Comment by JanKallman** [comment]
Fixed in version 5.1.1
